plugins/guests/centos/plugin.rb

```diff
@@ -0,0 +1,45 @@
+require "vagrant"
+
+module VagrantPlugins
+  module GuestCentos
+    class Plugin < Vagrant.plugin("2")
+      name "CentOS guest"
+      description "CentOS guest support."
+
+      guest(:centos, :linux) do
+        require_relative "guest"
+        Guest
+      end
+
+      guest_capability(:centos, :change_host_name) do
+        require_relative "../redhat/cap/change_host_name"
```

Comment:
centos should have pretty much all the same capabilities as redhat. I'm not sure if this is the ideal way to de-duplicate code in this case?

Reply:
Guests can inherit from a parent when the guest OS is a derivative of an existing distribution. A very simple example of this can be seen in the `ubuntu` guest which really just uses all the `debian` capabilities: https://github.com/hashicorp/vagrant/blob/master/plugins/guests/ubuntu/plugin.rb

Likewise, the `debian` guest inherits from the `linux` guest. The `linux` guest provides mostly generic capabilities, and then the `debian` guest will override specific capabilities which it provides customized implementations: https://github.com/hashicorp/vagrant/blob/master/plugins/guests/debian/plugin.rb